src/Microsoft.ML.ImageAnalytics/ImageType.cs

```diff
@@ -36,6 +38,19 @@ public override bool Equals(ColumnType other)
             return Width != tmp.Width;
         }
 
+        public override bool Equals(Object other)
+        {
+            var tmp = other as ColumnType;
+            if (tmp == null)
+                return false;
+            return Equals(tmp);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hashing.CombineHash(Height, Width);
```

Comment:
>Height, Width [](start = 39, length = 13)

I would prefer to combine the *hash* of these two items, rather than the items themselves. (Nevermind that the implementation of hash of integers literally just returns the value. :) It's the principle that counts.)

Reply:
I did that intentionally so I can get people commenting on it :), will change, thx

Reply:
Ah ha. So you are an activist raising awareness of the poor implementation of .NET's integer get hash method. You know, while there are far more serious problems in the world today, I nonetheless salute you for your service.

---
In reply to: [206717508](https://github.com/dotnet/machinelearning/pull/620#discussion_r206717508) [](ancestors = 206717508)